src/PrestaShopBundle/Translation/PrestaShopTranslatorTrait.php

```diff
@@ -59,18 +59,20 @@ public function trans($id, array $parameters = [], $domain = null, $locale = nul
             $locale = null;
         }
 
+        $translated = parent::trans($id, [], $this->normalizeDomain($domain), $locale);
+
         if ($this->shouldFallbackToLegacyModuleTranslation($id, $domain)) {
             return $this->translateUsingLegacySystem($id, $parameters, $domain, $locale);
         }
 
-        $translated = parent::trans($id, $isSprintf ? [] : $parameters, $this->normalizeDomain($domain), $locale);
+        $translated = isset($legacy) ? $this->replaceSpecialCharsWithLegacyFunctions($translated, $legacy) : $translated;
```

Comment:
If done this way the parameters don't get escaped, invalidating the main reason why the escape sequence is performed in the first place (to avoid code injection through unescaped parameters).